### DISCUSSION TITLE
Intricate Query in Resource List

### DISCUSSION THREAD

**Discussion by DavidPower** [filamentphp/filament#3878] [Help]
I have two basic resources: Customer and Transaction where a Customer can have one or more Transactions.

I want to create a single searchable, sortable, filterable list that combines the two resources with a couple of wrinkles:

1. Display the customer name and email (i.e. Customer resource) with the most recent transaction description and date (i.e. Transaction resource)
2. On the same list item, display the customer's LTV (i.e. sum of transactions.amount) across all the customer's transactions

I have a DB::select() query that generates the results I need. But I'm having challenges returning that in a form getTableQuery() will accept (i.e. as a Builder vs. array).

Can you offer a few clues on how I can make this work?

Thanks

**Comment by danharrin** [comment]
Hello,

> the most recent transaction description and date (i.e. Transaction resource)

https://laravel.com/docs/9.x/eloquent-relationships#has-one-of-many

Bear in mind that Laravel does not support sorting on this type of relationship yet, as far as I am aware

> On the same list item, display the customer's LTV (i.e. sum of transactions.amount) across all the customer's transactions

https://filamentphp.com/docs/2.x/tables/columns#aggregating-relationships

You should be able to completely avoid the need for a custom query

**Reply by DavidPower** [reply]
Thanks, Dan. 

I'll play around with this and see if I can get it to work.

Curious... (and perhaps a basic question): Can I create an Eloquent model based on a MySQL database view?

If so, it seems easier to create the view, then let Eloquent take care of the rest of the details (without complexity) inside Filament.

Thoughts?

**Reply by danharrin** [reply]
I'm not sure, you'll probably have to research that one yourself
